docs revamp: Plan IAM strategy (#62582)
grafana/grafana: docs/sources/auth/planning/index.md

---
title: Planning IAM integration strategy
menuTitle: IAM integration
description: Considerations and .
aliases:
  - /docs/mimir/latest/old-architecture/
  - docs/sources/auth/planning/
weight: 200
keywords:
  - IdP
  - IAM
  - Auth
---

> # Planning IAM integration strategy

Reviewer: ```suggestion
# Planning your IAM integration strategy
```

> The following documentation is meant to shed light on the different authorization and authentications strategies available in Grafana. By doing preparation before implementation, the user will be able to decide which integration strategy suits best for their needs.

Reviewer: ```suggestion
This topic describes the decisions you should make when using an identity and access management (IAM) provider to manage access to Grafana. When you prepare before you implement, you can decide which integration strategy best meets your needs.
```

Author: Changed.

Identity and Access Management (IAM) is needed to handle secure access to [Grafana resources]. Integrating Grafana with the proper solution is a key step to allow user access to sensitive data and resources, and to simplify user management and authentication.

## Benefits of having an IAM integration

> Consider the following benefits for IAM integration strategy with Grafana.

Reviewer: ```suggestion
Integrating with an IAM provider provides the following benefits:
```

> 1. **User management**: By providing Grafana access to your current user database, you eliminate the overhead of replicating any user information and instead have a centrilized user management for user's roles and permissions to Grafana resources.

Reviewer: ```suggestion
- **User management**: By providing Grafana access to your current user database, you eliminate the overhead of replicating user information and instead have centralized user management for users' roles and permissions to Grafana resources.
```

1. **Better security**: IAM solutions already provide advanced security features such as MFA, RBCA, and audit trails, which can help to improve the security of your Grafana installation.

> 1. **SSO**: By properly setting up Grafana with your current IAM solution will benefit user experience by accessing the Grafana instance with the same set of credentials as they use for everything else within the organization.

Reviewer: ```suggestion
- **SSO**: Properly setting up Grafana with your current IAM solution enables users to access Grafana with the same credentials they use for other applications.
```

> 1. **Scalability**: Adding, updating or removing users from your user database will reflect immediately at the Grafana instance.

Reviewer: ```suggestion
- **Scalability**: Adding, updating, or removing users from your user database is immediately reflected in Grafana.
```

> In order to plan an integration with Grafana, assess your organization's current needs, requirements and any existing IAM solutions being used. This should include which set of roles and permissions will be match to each type of user and which groups of users share the same set of permissions and access to shared resources.

Reviewer: It feels like we've moved out of benefits and into something else. Does the rest of the documentation expand on the points you've made in these two paragraphs?

Reviewer: I suppose we could put this content into the intro for now. Definitely feels like that's a better place for it at the moment. After it's there, I'll have additional thoughts.

After considering needs, chosen IAM solution and taking into consideration the security requirements, the last step is to test it out thoroughly before deploying it to a production environment.

> ## Where are my users?

Reviewer: ```suggestion
## Internal vs external users
```

> The first thing to consider is who are my users? Are the user exclusively within my organization or will they be outside of my organization?

Reviewer: ```suggestion
As a first step, determine where your Grafana users are located. Are the users located within your organization, or are they outside your organization?
```

Author: Changed.

> If the users are within my organization, this means that Grafana might be able to integrate with those users by providing connection with to the corresponding Identity Provider.

Reviewer: ```suggestion
If the users are within your organization, Grafana might be able to integrate with those users through an identify provider.
```

> If the users are outside of my organization, this means that Grafana needs to provide anonymous access, which is not enabled by default.

Reviewer: ```suggestion
If the users are outside your organization, you must provide anonymous access to Grafana, which is not enabled by default.
```

Reviewer: Should we provide a link to the enable anonymous access instructions? You can introduce that link as follows: "For information about enabling anonymous access, refer to [LINK]."

Author: Changed.

> ### Why do I need to organize the users?

Reviewer: @linoman It seems like this section and the sections that follow are related. Are you providing an overview of the benefits of organizing users here and then providing more detail about each of the options?

Author: The planning section will help the user understand that there should be some decision-making aspects to consider before moving on with the installation. This area could be improved by linking this with the references/terminology section. Since that section has not been started, I decided to add unresolved links as placeholders for now.

Do you think of a better way to develop this area for now?

### Choosing between teams and organizations

Grafana Teams and Grafana Organizations serve similar purposes in the Grafana platform. Both are designed to help manage and control access to resources, Teams provide a more focused approach for smaller groups or projects. Teams allow for easy collaboration and shared ownership of resources among members, making it simple to manage access and control on who has access to said resources.

Organizations, on the other hand, provide a higher level of management for multiple teams and resources, and are designed for larger enterprises or organizations with a complex setup or multiple and different business models.

That being said, Teams provide many of the same benefits as Organizations, and can be seen as a more streamlined and simplified approach to managing resources and access. As such, it is possible that in the future, Teams may eventually replace Organizations as the primary way to manage resources in Grafana.

> ## Do I have external systems?

Reviewer: ```suggestion
## Access to external systems
```

Author: Changed.

> Consider the need for [M2M] communications. If there's a system that needs to interact with Grafana, ensure that it has proper access for it.

Reviewer: ```suggestion
Consider the need for machine-to-machine (M2M) communications. If a system needs to interact with Grafana, ensure it has proper access.
```

Author: Thank you for the suggestion! I'm updating the text with it. However, I'm keeping the [M2M] as a placeholder for the terminology section.

> Here are some example scenarios:

Reviewer: ```suggestion
Consider the following scenarios:
```

Author: Changed.

> **IoT devices monitoring**: Sensors and actuators might want to ingest information to Grafana in an automatic way. Consider the security implications of a shared security access for all the devices or an individual access to each of them.

Reviewer: @linoman sensors and actuators can ingest information to Grafana or from Grafana? Or is it that Grafana ingests information from sensors?

Author: Reworded to avoid confusion.

> **Network monitoring**: Having a distributed systems architecture performance reported back to Grafana might provide insight about bottlenecks and even trigger alerts in the need of a timely intervention.

Reviewer: ```suggestion
**Network monitoring**: Having distributed systems architecture performance reported back to Grafana can provide insight into bottlenecks and trigger alerts that should be resolved promptly.
```

Author: Changed.

> These are just a few examples of how Grafana can be used in M2M scenarios. The platform is highly flexible and can be used in a variety of other M2M applications, making it a powerful tool for organizations looking to gain insights into their systems and devices.

Reviewer: @linoman , I would add one more scenario. Users like things that come in 3s. Also, do we have any blog posts that we can link the reader to for each of these scenarios? They might want to read up on a use case.

Author: Changed.

---

> title: Planning IAM integration strategy

Reviewer: ```suggestion
title: Plan your IAM integration strategy
```

> menuTitle: IAM integration

Reviewer: ```suggestion
menuTitle: Plan your integration
```

> description: Considerations and .

Reviewer 2: Looks like the end of the sentence is missing?

Author: 🤦 My bad. I have fixed this.

aliases:

>   - /docs/mimir/latest/old-architecture/

Reviewer 2: Should this Mimir link alias to this page?

  - docs/sources/auth/planning/
weight: 200
keywords:
  - IdP
  - IAM
  - Auth
---

> # Planning your IAM integration strategy

Reviewer: ```suggestion
# Plan your IAM integration strategy
```

> This topic describes the decisions you should make when using an Identity and Access Management (IAM) provider to manage access to Grafana. IAM ensures that users have secure access to sensitive data and [Grafana resources], simplifying user management and authentication.

Reviewer: Are you adding a link to [Grafana resources]?

## Benefits of integrating with an IAM provider

Integrating with an IAM provider provides the following benefits:

- **User management**: By providing Grafana access to your current user database, you eliminate the overhead of replicating user information and instead have centralized user management for users' roles and permissions to Grafana resources.

- **Security**: IAM solutions provide advanced security features such as MFA, RBCA, and audit trails, which can help to improve the security of your Grafana installation.

- **SSO**: Properly setting up Grafana with your current IAM solution enables users to access Grafana with the same credentials they use for other applications.

- **Scalability**: Adding, updating, or removing users from your user database is immediately reflected in Grafana.

In order to plan an integration with Grafana, assess your organization's current needs, requirements and any existing IAM solutions being used. This should include which set of roles and permissions will be match to each type of user and which groups of users share the same set of permissions and access to shared resources.

> After considering needs, chosen IAM solution and taking into consideration the security requirements, the last step is to test it out thoroughly before deploying it to a production environment.

Reviewer: ```suggestion
After considering your needs and security requirements and choosing an IAM solution, you should thoroughly test your solution before deploying it to a production environment.
```

## Internal vs external users

As a first step, determine where your Grafana users are located. Are the users located within your organization, or are they outside your organization?

If the users are within your organization, Grafana might be able to integrate with those users through an identify provider.

> If the users are outside your organization, you must provide anonymous access to Grafana, which is not enabled by default.

Reviewer 2: I think this depends on the use case. You could also create accounts using basic auth for the users outside the organisation. Or you could share dashboards using public dashboards if that satisfies the use case.

I think we need to be careful when advertising anonymous access, as it is not great security wise and is not a feature that we're investing in much.

For information about enabling anonymous access, refer to the [documentation](../../setup-grafana/configure-security/configure-authentication/_index.md#anonymous-authentication)

## Ways to organize users

> Organize users in subgroups that are sensible to the organization. These are some examples:

Reviewer: ```suggestion
We recommend organizing users into subgroups in a way that makes sense to your business. Consider the following approaches when you organize users.
```

- **Security**: Different groups of users or customers should only have access to their intended resources.
- **Simplicity**: Reduce the scope of dashboards and resources available.
- **Cost attribution**: Track and bill costs to their customers, departments, or divisions.

> - **Customization**: Each group of users could have a personalized experience like different dashboards or theme colours.

Reviewer 2: Do we support different theme colours for different teams?

Author: Yes, you can. [Here](https://grafana.com/docs/grafana/latest/administration/organization-preferences/#change-grafana-ui-theme) you can read more about it.

### Users in Grafana Teams

You can organize users into [Teams] and assign them roles and permissions reflecting the current organization. For example, instead of assigning five users access to the same dashboard, you can create a team that consists of those users and assign dashboard permissions to the team. A user can belong to multiple teams.

A user can be a Member or an Administrator for a given team. Members of a team inherit permissions from the team, but they cannot edit the team itself. Team Administrators can add members to a team and update its settings, such as the team name, team members' team roles, UI preferences, and the default dashboard to be displayed upon login for the team members.

Teams are a perfect solution for working with a small subset of users. Teams can share resources with other teams.

> ### Users in Grafana Organizations

Reviewer 2: Let's chat about orgs, there are some things I would reword slightly:
* orgs don't really isolate users from resources, but rather isolate the resources themselves;
* we could reword the reasoning for suggesting teams over orgs, as it's not just scalability.

---
title: Planning IAM integration strategy
menuTitle: IAM integration
description: Considerations and .
aliases:
  - /docs/mimir/latest/old-architecture/
  - docs/sources/auth/planning/
weight: 200
keywords:
  - IdP
  - IAM
  - Auth
---

# Planning your IAM integration strategy

> This topic describes the decisions you should make when using an identity and access management (IAM) provider to manage access to Grafana. When you prepare before you implement, you can decide which integration strategy best meets your needs.

Reviewer: ```suggestion
This topic describes the decisions you should make when using an identity and access management (IAM) provider to manage access to Grafana. When you prepare before you implement, you determine which integration strategy best meets your needs.
```

Author: Changed!

> Identity and Access Management (IAM) is needed to handle secure access to [Grafana resources]. Integrating Grafana with the proper solution is a key step to allow user access to sensitive data and resources, and to simplify user management and authentication.

Reviewer: ```suggestion
IAM ensures that users have secure access to sensitive data and [Grafana resources], and simplifies user management and authentication.
```

Author: Removed

> ## Benefits of having an IAM integration

Reviewer: ```suggestion
## Benefits of integrating with an IAM provider
```

Author: Changed

Integrating with an IAM provider provides the following benefits:

- **User management**: By providing Grafana access to your current user database, you eliminate the overhead of replicating user information and instead have centralized user management for users' roles and permissions to Grafana resources.

- **Security**: IAM solutions provide advanced security features such as MFA, RBCA, and audit trails, which can help to improve the security of your Grafana installation.

- **SSO**: Properly setting up Grafana with your current IAM solution enables users to access Grafana with the same credentials they use for other applications.

- **Scalability**: Adding, updating, or removing users from your user database is immediately reflected in Grafana.

In order to plan an integration with Grafana, assess your organization's current needs, requirements and any existing IAM solutions being used. This should include which set of roles and permissions will be match to each type of user and which groups of users share the same set of permissions and access to shared resources.

> After considering needs, chosen IAM solution and taking into consideration the security requirements, the last step is to test it out thoroughly before deploying it to a production environment.

Reviewer: same with this bit. It's interesting, you do mention testing and deployment here. One idea is that the parent IAM landing page provides the framework for all of the content with a graphic that shows Planning --> implementation --> testing --> deployment (if this makes sense) --> maintenance. I have some thoughts.

## Internal vs external users

As a first step, determine where your Grafana users are located. Are the users located within your organization, or are they outside your organization?

If the users are within your organization, Grafana might be able to integrate with those users through an identify provider.

If the users are outside your organization, you must provide anonymous access to Grafana, which is not enabled by default.

For information about enabling anonymous access, refer to the [documentation](../../setup-grafana/configure-security/configure-authentication/_index.md#anonymous-authentication)

## Ways to organize users

> 

Reviewer: Every list requires an introduction sentence that describes what's in the list. For example: "Organizing users into groups provides the following benefits:"

If that intro sentence is accurate, then I would suggest changing the section heading to "Benefits of user groups" or something like that.

Author: Intro was added as requested.

> ### Why do I need to organize the users?

Reviewer: ```suggestion
## Ways to organize users
```

Author: Changed.

> 1. **Security**: Different groups of users or customers should only have access to their intended resources.

Reviewer: These do not need to be a numbered list because these aren't steps. You can use an unorderedlist, which is set off by a `-`. I'll add an example in a suggestion.

Author: Changed

> 1. **Simplicity**: Reduce scope of dashboards and resources available.

Reviewer: ```suggestion
- **Simplicity**: Reduce the scope of dashboards and resources available.
```

Author: Changed

1. **Cost attribution**: Track and bill costs to their customers, departments, or divisions.

> 1. **Customization**: Each team could have a personalized experience like different dashboards or theme colours.

Reviewer: These look like benefits of using teams? Or at least two of them are benefits of using teams. Can these points be made more generic and remove the reference to teams specifically?

My thought is to have this section describe the benefits of using some mechanism to group users, and then you'd have two subsections that describe teams and another section that describes organizations. And then one final section that describes which approach to use.

Does that makes sense?

Author: It does! I removed references to teams and called them groups of users.

> ### Users in Grafana Teams

Author: A lot of references will be corrected and changed when the development of the Reference section is on its way. This will allow updating a lot of placeholders for proper doc references.

> It makes sense to organize users in [Teams] in order to assign them roles and permissions reflecting the current organization. For example, instead of assigning five users access to the same dashboard, you can create a team that consists of those users and assign dashboard permissions to the team. A user can belong to multiple teams.

Reviewer: ```suggestion
You can organize users into [Teams] and assign them roles and permissions reflecting the current organization. For example, instead of assigning five users access to the same dashboard, you can create a team that consists of those users and assign dashboard permissions to the team. A user can belong to multiple teams.
```

Author: Changed.

> A user can be a Member or an Administrator for a given team. Members of a team inherit permissions from the team, but they cannot edit the team itself. Team Administrators can add members to a team and update its settings, such as the team name, team member’s team roles, UI preferences, and the default dashboard to be displayed upon login for the team members.

Reviewer: ```suggestion
A user can be a Member or an Administrator for a given team. Members of a team inherit permissions from the team, but they cannot edit the team itself. Team Administrators can add members to a team and update its settings, such as the team name, team members’ team roles, UI preferences, and the default dashboard to be displayed upon login for the team members.
```

Author: Changed.

> Teams is a perfect solution to work with a small subset of users. Teams can share resources among other teams.

Author: Changed.

> ### Users in Grafana Organizations

Reviewer: As with the teams, there is a bunch of org content in the grafana docs. Maybe you can add links?

> [Grafana Organizations] was born as a concept to isolate users from dashboards and datasources by having multiple organizations under a single instance of Grafana. This means that users under different organizations won't share any resources such as dashboards, folders, and datasources.

Reviewer: ```suggestion
[Grafana Organizations] isolate users from dashboards and data sources by having multiple organizations under a single instance of Grafana. This means users under different organizations won't share any resources such as dashboards, folders, and data sources.
```

Author: Changed

> Grafana Organizations provide a measure of isolation within Grafana by default. The intention is to present different user experiences, which give the sense of different instances of Grafana within a single instance. However, **we recommend aginast Grafana Organizations** because they lack the scalability of [Folders].

Reviewer: ```suggestion
By default, organizations provide a measure of isolation within Grafana. They can be used to present different user experiences, which gives the appearance that there are multiple instances of Grafana within a single instance. However, because they lack the scalability of [Folders], we do not recommend using organizations as a way to group users.
```

Author: Changed

### Choosing between teams and organizations

> [Grafana Teams] and Grafana Organizations serve similar purposes in the Grafana platform. Both are designed to help manage and control access to resources, Teams provide a more focused approach for smaller groups or projects. Teams allow for easy collaboration and shared ownership of resources among members, making it simple to manage access and control on who has access to said resources.

Reviewer: ```suggestion
[Grafana teams] and Grafana organizations serve similar purposes in the Grafana platform. While both are designed to help manage and control access to resources, teams provide a more focused approach for smaller groups or projects. Teams also enable collaboration and shared ownership of resources among members, making it simple to manage access and control who has access to which resources.
```

Author: Changed

> Organizations, on the other hand, provide a higher level of management for multiple teams and resources, and are designed for larger enterprises or organizations with a complex setup or multiple and different business models.

Reviewer: ```suggestion
In contrast, organizations provide a higher level of management for multiple teams and resources. They are designed for larger enterprises or organizations with a complex setup or multiple or different business models.
```

Author: Changed

> That being said, Teams provide many of the same benefits as Organizations, and can be seen as a more streamlined and simplified approach to managing resources and access. As such, it is possible that in the future, Teams may eventually replace Organizations as the primary way to manage resources in Grafana.

Reviewer: I think you remove this content because it doesn't say anything you haven't already said, and it makes a promise about the future, which we should not do in tech docs. Talking about what might happen in the future is a maintenance issue for docs.

Author: Removed

## Access to external systems

> Consider the need for machine-to-machine [M2M] communications. If a system needs to interact with Grafana, ensure it has proper access.

Reviewer: is M2M supposed to be a linked? Or are you introducing it as an acronym such as (M2M)?

Author: This will be referenced to the Glossary

> # Plan IAM integration strategy

Reviewer: You'll need to add the front matter to this Markdown file. See guidelines here: https://grafana.com/docs/writers-toolkit/writing-guide/front-matter/

The following documentation is meant to shed light on the different authorization and authentications strategies available in Grafana. By doing preparation before implementation, the user will be able to decide which integration strategy suits best for their needs.

## Where are my users?

The first thing to consider is who are my users? Are the user exclusively within my organization or will they be outside of my organization?

If the users are within my organization, this means that Grafana might be able to integrate with those users by providing connection with to the corresponding Identity Provider.

If the users are outside of my organization, this means that Grafana needs to provide anonymous access, which is not enabled by default.

> ### 🚧 How are my users organized?

Reviewer 2: I'm guessing this would be a table of contents (`{{< section >}}`) and each of these sections would have an individual page, right?

I think it's useful to have it in this format for now, so we can see what remains to be done, but just wanted to check what your vision was :)

Author: Sorry for replying late to this comment. Yes! The idea of this section is to introduce the user to the different options. However, a more in-detail page will talk about specific Teams/Orgs/etc and used a reference in this section.

### 🚧 Users in teams

### 🚧 Users in organizations

### 🚧 Choosing between teams and organizations

## 🚧 Do I have external systems?

### 🚧 Service Accounts

> ### 🚧 Personal access tokens

Reviewer 2: We currently don't have them, so this will probably be a future entry 😄

> ### 🚧 API keys

Reviewer: @linoman , just curious. Are you also addressing terminology in this topic or will terminology be its own topic? The name of this section (API keys) looks to me like you are going to define what they are.

## 🚧 How to work with roles?

### 🚧 What are permissions?

### 🚧 What are roles?

> ### 🚧 Grafana roles vs RBAC: Which one is for me?

Reviewer 2: Grafana roles (Viewer, Editor, Admin, Server Admin) are also role-based access control roles. So I think this sounds slightly odd. Maybe having separate sections (something like `Managing basic roles` and `RBAC role assignments`) would make more sense 🤔
